### ISSUE TITLE
Crash related to WeatherSimulationSnowAccum

### ISSUE THREAD

**Issue by Squiddie82** [anegostudios/VintageStory-Issues#908]
**Game Version:** 1.15.0-pre.1
**Platform:** Windows
**Modded:** No

### Description
When walking around the world, the game will randomly crash. I've had this happen to me twice now. First when I first began the game, I was walking around collecting grass. Second, on the next day I played it crashed while walking home to my house at night. I find it strange that the crash references snow accumulation as the season is currently Spring and there is no visible snow around.

### How to reproduce
Walk around the world until it happens, I'm not sure what causes it.

### Expected behavior
For the game not to crash.

### Logs
```
Running on 64 bit Windows with 8 GB RAM 
Version: v1.15.0-pre.1 (Unstable)5/26/2021 11:36:47 AM: Critical error occurred
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Common.WorldChunk.GetLocalBlockAtBlockPos(IWorldAccessor world, BlockPos position)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayer(SnowAccumSnapshot sumsnapshot, Boolean ignoreOldAccum, IServerMapChunk mc, Vec2i chunkPos, IWorldChunk[] chunksCol)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.GetSnowUpdate(WeatherSimulationRegion simregion, IServerMapChunk mc, Vec2i chunkPos, IWorldChunk[] chunksCol)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayerOffThread(WeatherSimulationRegion simregion, IServerMapChunk mc, Vec2i chunkPos)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.onThreadTick()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
-------------------------------

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 5/26/2021 11:37:02 AM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.15.0.0, time stamp: 0x60aa9427
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x00007ffd6e93b4e9
Faulting process id: 0x6bc
Faulting application start time: 0x01d7522ee987069b
Faulting application path: C:\Users\emma8\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: unknown
Report Id: 2838f95b-1ef3-4722-8bbc-d55446edb92b
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 5/26/2021 11:36:58 AM, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Vintagestory.Common.WorldChunk.GetLocalBlockAtBlockPos(Vintagestory.API.Common.IWorldAccessor, Vintagestory.API.MathTools.BlockPos)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayer(Vintagestory.GameContent.SnowAccumSnapshot, Boolean, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i, Vintagestory.API.Common.IWorldChunk[])
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.GetSnowUpdate(Vintagestory.GameContent.WeatherSimulationRegion, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i, Vintagestory.API.Common.IWorldChunk[])
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.UpdateSnowLayerOffThread(Vintagestory.GameContent.WeatherSimulationRegion, Vintagestory.API.Common.IServerMapChunk, Vintagestory.API.MathTools.Vec2i)
   at Vintagestory.GameContent.WeatherSimulationSnowAccum.onThreadTick()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart()

 }
--------------
{ TimeGenerated = 5/24/2021 4:10:03 PM, Site = , Source = Windows Error Reporting, Message = Fault bucket 1498583595829563708, type 5
Event Name: CLR20r3
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.15.0.0
P3: 60aa9427
P4: VintagestoryLib
P5: 1.15.0.0
P6: 60aa9427
P7: 1407
P8: 2e
P9: System.NullReferenceException
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER611F.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER834E.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8468.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER8475.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER84C5.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_d879de9517d869d76b058602040a4b978bf85b_cdf018de_3bf4b79e-06f1-4b89-921b-eac118c81947

Analysis symbol: 
Rechecking for solution: 0
Report Id: f2e61462-366b-4341-92ec-19a3fc5f0b73
Report Status: 268435456
Hashed bucket: 0f9e7028cc8f4d1044cc09d72fa6113c
Cab Guid: 0 }
```

**Comment by Craluminum2413** [comment]
I think it is related with snow appearing on some blocks

**Comment by Craluminum2413** [comment]
@Squiddie82  Can you send a world seed? /seed

**Comment by Squiddie82** [comment]
The world seed is 1829935939

**Comment by Craluminum2413** [comment]
I think it is fixed in 1.15.0-pre.3

**Comment by radfast** [comment]
I believe this is fixed in `1.15-pre.3` (see #News channel on Discord).  If you still have the same issue please re-report it and we should give it high priority.
